Add New Ethos-U85 support to Arm aot_arm_compiler

robell opened the pull request:
Some changes to enable using the new compiler flow and backend for both Ethos-U55 and Ethos-U85.

This change means the run.sh flow uses the new compiler for Ethos-U55.

It enables Ethos-U85 to be targeted for compilation, but this feature isn't used in run.sh yet.

robell committed:
Improvements to the Arm aot_arm_compiler flow.

- Support for nn.modules in a seperate file
- Ability to select target IP for the ArmBackend
- Option to dump intermediate outputs for debug.

Signed-off-by: Rob Elliott <robert.elliott@arm.com>
Change-Id: I40cef76da593bd6f06bab067d3f0fc1ae31adc44

robell committed:
lintfix

robell commented:
@digantdesai can we push PR #5308, then i'll rebase and remove my commit which does much the same.

digantdesai commented on examples/arm/aot_arm_compiler.py:
```diff
@@ -149,6 +169,14 @@ def forward(self, x):
+        "--target",
+        action="store",
+        required=False,
+        default="ethos-u55-128",
```
```suggestion
        default="ethos-u55-128",
        choices=targets,
```

robell replied:
Ack.

digantdesai commented on examples/arm/aot_arm_compiler.py:
```diff
@@ -209,19 +251,51 @@ def forward(self, x):
     if args.delegate is True:
-        edge = edge.to_backend(
-            ArmPartitioner(
+        if args.target == "TOSA":
```
nit refctor as `def get_compile_spec(target: str)`?

robell replied:
Ack.

digantdesai reviewed (approved):
lgtm thanks.

digantdesai commented:
> @digantdesai can we push PR #5308, then i'll rebase and remove my commit which does much the same. and btw, the lintrunner fix isn't needed as that's the patch which will be reverted.

I will merge #5308 ASAP and then you can rebase.

robell committed:
tidy compilespec building into a function

Signed-off-by: Rob Elliott <robert.elliott@arm.com>

robell committed:
simplify function for lintrunner

Signed-off-by: Rob Elliott <robert.elliott@arm.com>

robell committed:
Fix missing quantize_io on U85 compile

Signed-off-by: Rob Elliott <robert.elliott@arm.com>

digantdesai commented:
Seems like U55 tests are failing.

robell committed:
Add Arm Ethos-U85 to aot_arm_compiler

With Ethos-U85 being enabled in the main ArmBackend and compile spec, this change
enables it from the command line for compilation in preperation for run.sh support
in newer examples.

Signed-off-by: Rob Elliott <robert.elliott@arm.com>
Change-Id: I33e44f000fb24bc70b468fbffbf547e9775a1132

robell commented:
@digantdesai that failure was a side-effect of the pin move so was resolved now this is rebased on top of #5346 unit test fixes . looks like all but meta internal-only has returned a result.

digantdesai commented:
Let me try to merge this ASAP
